bip-0085.mediawiki

```diff
@@ -360,36 +360,58 @@ OUTPUT
 * DERIVED ENTROPY=f7cfe56f63dca2490f65fcbf9ee63dcd85d18f751b6b5e1c1b8733af6459c904a75e82b4a22efff9b9e69de2144b293aa8714319a054b6cb55826a8e51425209
 * DERIVED PWD=_s`{TW89)i4`
 
-===RSA===
+===GPG Keys===
 
-Application number: 828365'
+Application number is dependant on the key type.
 
-The derivation path format is: <code>m/83696968'/828365'/{key_bits}'/{key_index}'</code>
+{|
+!OpenGPG Key Type
+!Application Number
+|-
+| RSA
+| 828365'
+|-
+| ECC(Curve25519)
+| 828366'
+|-
+| ECC(secp256k1)
+| 828367'
+|-
+| ECC(NIST)
+| 828368'
+|-
+| ECC(Brainpool)
+| 828369'
+|-
+|}
 
-The RSA key generator should use BIP85-DRNG as the input RNG function.
+The RSA key generator should use BIP85-DRNG as the input RNG function. Likewise, any ECC key types over 256bit should use BIP85-DRNG.
 
-===RSA GPG===
+====Primary Keys and Subkeys====
 
-Keys allocated for RSA-GPG purposes use the following scheme:
+Keys allocated for GPG purposes use the following scheme:
 
- - Main key <code>m/83696968'/828365'/{key_bits}'/{key_index}'</code>
- - Sub keys:  <code>m/83696968'/828365'/{key_bits}'/{key_index}'/{sub_key}'</code>
+ - Primary key <code>m/83696968'/{key_type}'/{key_bits}'/{key_index}'</code>
+ - Sub keys:  <code>m/83696968'/{key_type}'/{key_bits}'/{key_index}'/{sub_key}'</code>
 
     - key_index is the parent key for CERTIFY capability
     - sub_key <code>0'</code> is used as the ENCRYPTION key
     - sub_key <code>1'</code> is used as the AUTHENTICATION key
     - sub_key <code>2'</code> is usually used as SIGNATURE key
 
-Note on timestamps:
+Additional subkeys can be added to a primary key, including keys of a different key type, following the role pattern defined above, but the key_index MUST be incremented with each subkey.
```

Comment:
As for mixing types I would discourage that unless there's a really good reason. For one thing are we nesting paths at that point? A worked example of mixed subkeys would help.

Reply:
> the use-case for this is basically scenarios where someone may want a strong and long-lived primary key that for their identity, but use different keys for the actual signing and stuff. (Whether due to limitations on smartcards, etc) For example, someone may have an ED25119 primary key and have subkeys use something like P256 or RSA subkeys for compatibility reasons.

reasonable enough. at that point why not turtles all the way down, e.g. the following or something like it?

```
{same_app_number_for_all}/
{key_type}/{key_bits}/key_index}/
{key_type}/{key_bits}/sub_key_index}/
...
```